soil_id/global_soil.py

```diff
@@ -1159,154 +1208,226 @@ def rankPredictionGlobal(
-                "score_data_loc": "",
-                "rank_data_loc": "",
-                "score_data": "",
-                "rank_data": "",
-                "score_loc": round(row.distance_score_norm, 3),
-                "rank_loc": row.Rank_Loc,
-                "componentData": row.missing_status,
-            }
-        else:
-            rank_entry = {
-                "name": row.compname.capitalize(),
-                "component": row.compname_grp.capitalize(),
-                "mapunitID": row.mukey,
-                "componentID": row.cokey,
-                "score_data_loc": round(row.Score_Data_Loc, 3),
-                "rank_data_loc": row.Rank_Data_Loc,
-                "score_data": round(row.Score_Data, 3),
-                "rank_data": row.Rank_Data,
-                "score_loc": round(row.distance_score_norm, 3),
-                "rank_loc": row.Rank_Loc,
-                "componentData": row.missing_status,
-            }
-        rank_list.append(rank_entry)
+    # Construct the output format
+    Rank = [
+        {
+            "name": row.compname.capitalize(),
+            "component": row.compname_grp.capitalize(),
+            "componentID": row.cokey,
+            "score_data_loc": (
+                "" if row.missing_status == "Location data only" else round(row.Score_Data_Loc, 3)
+            ),
+            "rank_data_loc": (
+                "" if row.missing_status == "Location data only" else row.Rank_Data_Loc
+            ),
+            "score_data": (
+                "" if row.missing_status == "Location data only" else round(row.Score_Data, 3)
+            ),
+            "rank_data": "" if row.missing_status == "Location data only" else row.Rank_Data,
+            "score_loc": round(row.distance_score, 3),
+            "rank_loc": row.Rank_Loc,
+            "componentData": row.missing_status,
+        }
+        for _, row in D_final_loc.iterrows()
+    ]
 
-    # Setting up the return data structure
-    model_version = 3
-    metadata = {
-        "location": "global",
-        "model": "v3",
-        "dataCompleteness": {"score": data_completeness, "text": text_completeness},
+    output_data = {
+        "metadata": {
+            "location": "global",
+            "model": "v2",
+        },
+        "soilRank": Rank,
     }
 
-    result = {"metadata": metadata, "soilRank": rank_list}
-
-    # Save data if record_id is provided
-    if record_id:
-        save_rank_output(record_id, model_version, json.dumps(result))
-
-    return result
+    return output_data
 
 
 ##################################################################################################
 #                                          getSoilGridsGlobal                                    #
 ##################################################################################################
-def getSoilGridsGlobal(lon, lat, plot_id=None):
 
-    # Call soildgrids API
+
+def sg_list(lon, lat):
```

Comment:
suggestion: rename for clarity
```suggestion
def soil_grids_list(lon, lat):
```